core/src/cats/ddl/creates/postgresql.sql

```diff
@@ -597,6 +597,75 @@ WHERE (db_mb IS NOT NULL AND db_mb > 0)
    OR (vm_mb IS NOT NULL AND vm_mb > 0)
 ;
 
-
+-- lstat_decode part
+-- This function emulate in plpgsql language the C++ code of FromBase64 function in core/src/lib/base64.cc
+create or replace function bareos_frombase64(field text) returns bigint as $$
+ declare
+  i integer default 0;
+  res bigint default 0;
+  rf text default '';
+  base64 text default 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';
+  z integer := char_length(field); -- minus 1s
+ begin
+  rf := substring(field from 0 for 1);
+  -- skip negative
+  if rf = '-' then i := i+1; end if;
+  rf := substring(field from i for 1);
+  while  ( rf != ' ' ) and ( i <= z ) loop
+    res := res << 6;
+    res := res + position(rf in base64) - 1;
+    i := i + 1;
+    rf := substring(field from i for 1);
+  end loop;
+  return res;
+end
+$$ language 'plpgsql' immutable parallel safe;
+
+-- This function emulate in pgpgsql language the C++ code of DecodeStat function in core/src/lib/attrib.cc
+-- LinkFI is decode here as bigint which differs from the C code.
+create or replace function decode_lstat(lst text)
+ returns table(
+   st_dev bigint,
+   st_ino bigint,
+   st_mode bigint,
+   st_nlink bigint,
+   st_uid bigint,
+   st_gid bigint,
+   st_rdev bigint,
+   st_size bigint,
+   st_blksize bigint,
+   st_blocks bigint,
+   st_atime bigint,
+   st_mtime bigint,
+   st_ctime bigint,
+   linkfi bigint,
+   st_flags bigint,
+   data bigint
+ ) as $$
+ declare
+  fields text[];
+ begin
+  fields = string_to_array(lst, ' '); -- minus 10s
```

Comment:
what does that comment mean?

Reply:
good catch I forget to remove them. will do